File: src/DesignPattern/Structural/Composite/Composite-go/composite_Test.go
package main

// TestComponsite ...
func TestComponsite() {
	var root Component = &Composite{data: 0}
	root.Add(&Leaf{data: 12})
	root.Add(&Leaf{data: 16})
	var comp = &Composite{data: 99}
	comp.Add(&Leaf{data: 55})
	root.Add(comp)

	root.PrintInfo()
	root.Increase(10)
	root.PrintInfo()
}

func main() {
	TestComponsite()
}

File: src/DesignPattern/Structural/Composite/Composite-ts/composite.test.ts
import { Composite, Component, Leaf } from "./composite"


const root: Component = new Composite(0);
root.Add(new Leaf(1))
root.Add(new Leaf(2))
let compl = () => {
    let p = new Composite(12);
    p.Add(new Leaf(19));
    p.Add(new Leaf(67));
    return p;
}
root.Add(compl());

root.Print();
root.Increase();
root.Print();